src/main/java/org/folio/rest/impl/UserGroupAPI.java

```diff
@@ -40,323 +32,176 @@
-        log.error(e.getMessage(), e);
-        asyncResultHandler.handle(io.vertx.core.Future.succeededFuture(
-                PostGroupsResponse.respond500WithTextPlain(
-                messages.getMessage(lang, MessageConsts.InternalServerError))));
-      }
-    });
-
+      });
   }
 
   @Validate
   @Override
   public void getGroupsByGroupId(String groupId, String lang, Map<String, String> okapiHeaders,
-      Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
-
-    vertxContext.runOnContext(v -> {
-      try {
-        System.out.println("sending... getGroupsByGroupId");
-
-        Criterion c = new Criterion(
-          new Criteria().addField(ID_FIELD_NAME).setJSONB(false).setOperation("=").setValue("'"+groupId+"'"));
-
-        PostgresClientUtil.getInstance(vertxContext, okapiHeaders).get(GROUP_TABLE, Usergroup.class, c, false,
-            reply -> {
-              try {
-                if(reply.succeeded()){
-                  @SuppressWarnings("unchecked")
-                  List<Usergroup> userGroup = reply.result().getResults();
-                  if(userGroup.isEmpty()){
-                    asyncResultHandler.handle(io.vertx.core.Future.succeededFuture(GetGroupsByGroupIdResponse
-                      .respond404WithTextPlain(groupId)));
-                  }
-                  else{
-                    asyncResultHandler.handle(io.vertx.core.Future.succeededFuture(GetGroupsByGroupIdResponse
-                      .respond200WithApplicationJson(userGroup.get(0))));
-                  }
-                }
-                else{
-                  log.error(reply.cause().getMessage(), reply.cause());
-                  if(isInvalidUUID(reply.cause().getMessage())){
-                    asyncResultHandler.handle(io.vertx.core.Future.succeededFuture(GetGroupsByGroupIdResponse
-                      .respond404WithTextPlain(groupId)));
-                  }
-                  else{
-                    asyncResultHandler.handle(io.vertx.core.Future.succeededFuture(GetGroupsByGroupIdResponse
-                      .respond500WithTextPlain(messages.getMessage(lang, MessageConsts.InternalServerError))));
-                  }
-                }
-              } catch (Exception e) {
-                log.error(e.getMessage(), e);
-                asyncResultHandler.handle(io.vertx.core.Future.succeededFuture(GetGroupsByGroupIdResponse
-                  .respond500WithTextPlain(messages.getMessage(lang, MessageConsts.InternalServerError))));
-              }
-        });
-      } catch (Exception e) {
-        log.error(e.getMessage(), e);
-        asyncResultHandler.handle(io.vertx.core.Future.succeededFuture(GetGroupsByGroupIdResponse
-          .respond500WithTextPlain(messages.getMessage(lang, MessageConsts.InternalServerError))));
-      }
-    });
+    Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
+
+    // PgUtil.getById .. returns 500 where the below returns 404
```

Comment:
Returning 500 for an id string that isn't in valid UUID format is reasonable because the client most likely has mixed up something.
I suggest to change the unit test to both accept 404 and 500 for an invalid UUID.
And to use PgUtil.getById.

Reply:
I don't want to change behavior. If you pass something in as a client it's CLIENT ERROR (4XX).

Reply:
That's true, thanks for the clarification.

Opened https://issues.folio.org/browse/RMB-400 "PgUtil should return 400 (not 500) on invalid UUID id"